rfcs/0046-platform-support-tiers.md

```diff
@@ -0,0 +1,121 @@
+---
+feature: platform_support_tiers
+start-date: 2019-04-28
+author: Michael Raskin
+co-authors:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Document the way to describe the level of support a platform can expect in
+Nixpkgs.
+
+# Motivation
+[motivation]: #motivation
+
+When adding a new platform, be it a new C library option, a new
+cross-compilation target configuration or a new CPU architecture, there is a
+discussion of support expectations and maintenance burden. Having a documented
+vocabulary to describe the expectations and documented precedents should make
+such discussions more efficient.
+
+# Detailed design
+[design]: #detailed-design
+
+## Platform elements
+
+* CPU architecture (possibly: subarchitecture, optional features)
+* OS kernel
+* C compiler
+* C library
+* NixOS/non-NixOS global layout, in case of Linux with glibc
+* Global linking options
```

Comment:
Package sets: Gnome packages, language specific package, wayland packages, ...

Reply:
I would say with the caption of «platfom support» these go into the first point of «questions to answer» — «what fraction of packages works».

Indeed, my gist https://gist.github.com/7c6f434c/431c8233793571cad313e77356d2f7db does have some sketches w.r.t. package support level and some Nix-related kinds of support levels both for platforms and for Nix features, but I think there are less misunderstandings there.

(So we agree: different scope, let's start with platforms)